File: app/src/main/java/net/tlalka/android/fiszki/domain/utils/ValidUtils.java
package net.tlalka.android.fiszki.domain.utils;

import java.util.List;
import java.util.Map;

public class ValidUtils {

    private ValidUtils() {
    }

    public static boolean isNull(Object value) {
        return value == null;
    }

    public static boolean isNotNull(Object value) {
        return value != null;
    }

    public static boolean isTrue(Boolean value) {
        return isNotNull(value) && value;
    }

    public static boolean isFalse(Boolean value) {
        return isNotNull(value) && !value;
    }

    public static boolean isEmpty(String value) {
        return isNotNull(value) && value.length() == 0;
    }

    public static boolean isEmpty(List value) {
        return isNotNull(value) && value.isEmpty();
    }

    public static boolean isEmpty(Map value) {
        return isNotNull(value) && value.isEmpty();
    }

    public static boolean isNotEmpty(String value) {
        return isNotNull(value) && value.length() > 0;
    }

    public static boolean isNotEmpty(List value) {
        return isNotNull(value) && !value.isEmpty();
    }

    public static boolean isNotEmpty(Map value) {
        return isNotNull(value) && !value.isEmpty();
    }
}

File: app/src/main/java/net/tlalka/android/fiszki/domain/utils/ValidUtils.kt
package net.tlalka.android.fiszki.domain.utils

object ValidUtils {

    fun isNull(value: Any?): Boolean {
        return value == null
    }

    fun isNotNull(value: Any?): Boolean {
        return value != null
    }

    fun isTrue(value: Boolean?): Boolean {
        return value == true
    }

    fun isFalse(value: Boolean?): Boolean {
        return value == false
    }

    fun isEmpty(value: String?): Boolean {
        return value?.length == 0
    }

    fun isEmpty(value: List<*>?): Boolean {
        return value?.isEmpty() ?: false
    }

    fun isEmpty(value: Map<*, *>?): Boolean {
        return value?.isEmpty() ?: false
    }

    fun isNotEmpty(value: String?): Boolean {
        return value?.isNotBlank() ?: false
    }

    fun isNotEmpty(value: List<*>?): Boolean {
        return value?.isNotEmpty() ?: false
    }

    fun isNotEmpty(value: Map<*, *>?): Boolean {
        return value?.isNotEmpty() ?: false
    }
}
